### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by benr77**
## Contribution type

Improvements to plugin configuration documention

## Description of change

* Centralised common plugin options on a new page. 
* Linked to this page from each individual plugin page.
* Removed common options from each individual plugin's options list

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+.idea
```
Comment (corpsee) [inline .gitignore]: Directory `.idea` should be in your home dirirectory `.gitignore`.

File: .gitignore [hunk 2]
```diff
@@ -23,4 +25,4 @@
 /public/.htaccess
 
 /tests/runtime
-!/tests/runtime/.gitkeep
\ No newline at end of file
+!/tests/runtime/.gitkeep
```
Comment (corpsee) [inline .gitignore]: Why? :)

**Comment by corpsee** [comment]
@benr77 Could you rebase your branch ahead branch `release-1.0` instead of `master` please?

File: .gitignore (hunk 2 above)
Comment (benr77) [reply .gitignore]: This is already in `master` !

File: .gitignore (hunk 1 above)
Comment (benr77) [reply .gitignore]: Done

File: docs/en/plugin_common_options.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+General Plugin Options
+=================
+
+The following options are available for all plugins. 
+
+* **allow_failures** [bool, optional] - If true, allow the build to succeed even if this plugin fails.
+* **directory** [string, optional] - This option lets you specify the tests directory to run.
```
Comment (garas) [inline docs/en/plugin_common_options.md]: "This option lets you specify the directory this plugin to run on."

Plugin function is not always to test something.

File: docs/en/plugins/php_loc.md [hunk 4]
```diff
@@ -6,12 +6,12 @@ Runs [PHPLoc](https://github.com/sebastianbergmann/phploc) against your project
...
 
+See also [Common Plugin Configuration Options](../plugin_common_options.md).
+
 ### Options
 
-* **directory** - Optional - The directory in which phploc should run. 
-* **binary_name** [string|array, optional] - Allows you to provide a name of the binary
-* **binary_path** [string, optional] - Allows you to provide a path to the binary
-* **ignore** 
+None, except [Common Plugin Configuration Options](../plugin_common_options.md)
```
Comment (garas) [inline docs/en/plugins/php_loc.md]: I don't like these _See also..._ and _None, except..._ with same link here and for few other plugins.

I would leave one of them.

**Comment by benr77** [comment]
Ok, so as @garas has pointed out, some of the plugins do not use some of the common options.

Would it be better to actually list the common options on each plugin page, instead of linking to them? I was trying to reduce duplication but it seems there are differences. This approach would address both of @garas 's queries.

File: docs/en/plugins/phar.md [hunk 5]
```diff
@@ -6,11 +6,12 @@ Allows you to create a [Phar](http://php.net/manual/en/book.phar.php) archive fr
 Configuration
 -------------
 
+See also [Common Plugin Configuration Options](../plugin_common_options.md).
+
 ### Options
 
-* **directory**: `phar` output directory. Default: `%buildpath%`;
```
Comment (garas) [inline docs/en/plugins/phar.md]: Plugin like this pretty different `directory` purpose. In such cases it could be left.

Comment (garas) [reply docs/en/plugins/phar.md]: Another option is to use `directory` only as input directory and add new `output_path` line  like other plugins.

But this is idea for separate PR.

**Comment by corpsee** [comment]
> Ok, so as @garas has pointed out, some of the plugins do not use some of the common options.
> 
> Would it be better to actually list the common options on each plugin page, instead of linking to them? I was trying to reduce duplication but it seems there are differences. This approach would address both of @garas 's queries.

Ok.

File: docs/en/plugins/phar.md (hunk 5 above)
Comment (corpsee) [reply docs/en/plugins/phar.md]: @garas It is good idea for version `2.0.0` (because it is backward incompatible).

Comment (corpsee) [reply docs/en/plugins/phar.md]: We may add new option `output_path` as alternative in minor version `1.x` and deprecate `directory` for some plugins.

**Comment by corpsee** [comment]
@benr77 Can I merge your PR?

**Pull request closed without merging**